jboss/container/wildfly/launch/elytron/added/launch/elytron.sh

```diff
@@ -329,6 +335,37 @@ configure_security_domains() {
     configure_http_application_security_domains
     configure_ejb_application_security_domains
   fi
+  if [ -n "${ELYTRON_SECDOMAIN_NAME}" ]; then
+    validate_elytron_security_domains_env
+    if [ -z "${ELYTRON_SECDOMAIN_CORE_REALM}" ]; then
+      elytron_sec_domain="${ELYTRON_SECDOMAIN_NAME}"
+      configure_elytron_realm application-properties-"${ELYTRON_SECDOMAIN_NAME}"
+      configure_elytron_security_domain "${ELYTRON_SECDOMAIN_NAME}" application-properties-"${ELYTRON_SECDOMAIN_NAME}"
+    else
+      elytron_sec_domain="ApplicationDomain"
+    fi
+    configure_undertow_security_domain "${ELYTRON_SECDOMAIN_NAME}" "$elytron_sec_domain"
+    configure_ejb_security_domain "${ELYTRON_SECDOMAIN_NAME}" "$elytron_sec_domain"
+  fi
+}
+
+validate_elytron_security_domains_env() {
+  if [ -n "${ELYTRON_SECDOMAIN_CORE_REALM}" ]; then
+    if [ -n "${ELYTRON_SECDOMAIN_USERS_PROPERTIES}" ] || [ -n "${ELYTRON_SECDOMAIN_ROLES_PROPERTIES}" ]; then
+      log_error "When configuring an ELYTRON_SECDOMAIN, ELYTRON_SECDOMAIN_USERS_PROPERTIES and ELYTRON_SECDOMAIN_ROLES_PROPERTIES can't be used with ELYTRON_SECDOMAIN_CORE_REALM."
```

Comment:
Rather than 
```
log_error "When configuring..."
exit 1
```
As the configuration launch scripts run in separate sub-shells, this `exit` will not stop the full boot, it just exists this sub-shell. So as long as the CLI commands from the rest of this are something valid the server will still start.

Instead redirect to error file:
```
echo "When configuring..." >> "${CLI_SCRIPT_ERROR_FILE}"
return # or exit 1 is probably ok
```
When you echo to $CLI_SCRIPT_ERROR_FILE it picks that up before running the cli scripts. It checks that file, does log_error with the contents and calls exit1 (see processErrorsAndWarnings in openshift-common.sh). This way it exits properly and does not start the server